src/lib/helpers.ts

```diff
@@ -60,6 +61,39 @@ export function assertIsPlainObject(
   }
 }
 
+export function assertIsValidCodeVerifier(
+  verifier: unknown
+): asserts verifier is string {
+  assertIsString(verifier, "Invalid 'code_verifier' type");
+  if (!isValidPkceCodeVerifier(verifier)) {
+    throw new AssertionError({
+      message:
+        "Invalid 'code_verifier'. The verifier does not confirm with the RFC7636 spec. Ref: https://datatracker.ietf.org/doc/html/rfc7636#section-4.1",
+    });
+  }
+}
```

Comment:
This doesn't feel right. Unless I'm mistaken, type assertions are used to validate that a variable matches a given data type, but you are using it as a content validator.

I'd rather put the body of this function where it's being used (`src/lib/oauth2-service.ts`), and drop this function.